qcodes_contrib_drivers/drivers/ERAInstruments/erasynth.py

```diff
@@ -0,0 +1,900 @@
+
+        self.wifi_hotspot_password = Parameter(
+            name="wifi_hotspot_password",
+            instrument=self,
+            vals=validators.Strings(),
+            get_cmd=f"RA:{_CMD_TO_JSON_MAPPING['PEP1']}",
+            set_cmd="PEP1{}",
+        )
+        """Sets hotspot password for WiFi module."""
+
+        self.wifi_ip_address = Parameter(
+            name="wifi_ip_address",
+            instrument=self,
+            vals=validators.Strings(),
+            get_cmd=f"RA:{_CMD_TO_JSON_MAPPING['PEI']}",
+            set_cmd="PEI{}",
+        )
+        """Sets IP address for WiFi module."""
+
+        self.wifi_subnet_address = Parameter(
+            name="wifi_subnet_address",
+            instrument=self,
+            vals=validators.Strings(),
+            get_cmd=f"RA:{_CMD_TO_JSON_MAPPING['PEN']}",
+            set_cmd="PEN{}",
+        )
+        """Sets Subnet mask for WiFi module."""
+
+        self.wifi_gateway_address = Parameter(
+            name="wifi_gateway_address",
+            instrument=self,
+            vals=validators.Strings(),
+            get_cmd=f"RA:{_CMD_TO_JSON_MAPPING['PEG']}",
+            set_cmd="PEG{}",
+        )
+        """Sets default gateway for WiFi module."""
+
+        self.debug_messages_en = Parameter(
+            name="debug_messages_en",
+            instrument=self,
+            vals=validators.Strings(),
+            initial_cache_value=None,
+            val_mapping={True: "1", False: "0"},
+            set_cmd="PD{}",
+        )
+        """Enables/disables debug printing on the serial port."""
+
+        self.visa_handle.baud_rate = BAUDRATE
+        self.timeout(10)  # generous timeout to avoid disrupting measurements
+        self.connect_message()
+        self._prep_communication()
+
+    # ##################################################################################
+    # Public methods
+    # ##################################################################################
+
+    # Standard LO methods
+
+    def on(self) -> None:
+        """
+        Turns ON the RF output.
+        """
+        self.status(True)
+
+    def off(self) -> None:
+        """
+        Turns OFF the RF output.
+        """
+        self.status(False)
+
+    # Custom communication
+
+    def get_idn(self) -> Dict[str, str]:
+        models: Dict[str, str] = {"0": "ERASynth", "1": "ERASynth+", "2": "ERASynth++"}
+        d_status: Dict[str, str] = self.get_diagnostic_status()
+        return {
+            "vendor": "ERA Instruments",
+            "model": models[d_status["model"]],
+            "serial": d_status["serial_number"],
+            "firmware": d_status["em"],
+        }
+
+    def clear_read_buffer(self) -> None:
+        """
+        Clears the read buffer. The instrument often sends information we do not care
+        about in this driver. This function discards the entire read buffer by reading
+        it.
+        """
+        while self.visa_handle.bytes_in_buffer:
```

Comment:
looking into visa_handle in qcodes, it refers to MessageBasedResource class in visa.resources.messagebased.
I can't find `bytes_in_buffer` attribute for this class, and I don't where this attribute is defined previously. 
Am I missing something?

Reply:
It certainly worked while I was testing in a notebook

I guess it somehow routes to this:
https://pyvisa.readthedocs.io/en/latest/api/resources.html#pyvisa.resources.SerialInstrument.bytes_in_buffer

Reply:
OK. I wanted to make sure if everything is correct in the run time, then we can look into mypy thing

Reply:
I double checked this as well:

![Screenshot 2021-11-02 at 13 28 06](https://user-images.githubusercontent.com/31376402/139856163-a0b1ed6d-5be3-481b-8b84-bc854964599d.png)

![Screenshot 2021-11-02 at 13 27 30](https://user-images.githubusercontent.com/31376402/139856183-901b8908-b70a-410f-a610-5d875e379cc9.png)